projects/plugins/jetpack/extensions/blocks/markdown/renderer.js

```diff
@@ -21,8 +22,13 @@ const handleLinkClick = event => {
 	}
 };
 
-export default ( { className, source = '' } ) => (
-	<RawHTML className={ className } onClick={ handleLinkClick }>
+const getStyles = ( attributes = {} ) => {
+	const spacingProps = getSpacingClassesAndStyles( attributes );
```

Comment:
Should we check that `getSpacingClassesAndStyles` exists before we use it, given that it's still marked as experimental?

Reply:
> Should we check that getSpacingClassesAndStyles exists before we use it, given that it's still marked as experimental?

Thank you for the feedback, Jeremy! That's a good point.

I have added check in the new commit: https://github.com/Automattic/jetpack/pull/22429/commits/e367e3d0fc7c8ba383d50387784bd1486b2cab87:

```
const getStyles = ( attributes = {} ) => {
	const spacingProps = getSpacingClassesAndStyles?.( attributes );
	return spacingProps?.style ? spacingProps.style : {};
};
```

In case the `getSpacingClassesAndStyles()` doesn't exist, `spacingProps` will be undefined and `getStyles()` will return an empty object. Padding won't be applied and the app won't crash. This is what the user would see:

![Markup on 2022-01-21 at 18:36:30](https://user-images.githubusercontent.com/25105483/150574546-1f42e99b-ab28-4c7b-97ac-07b280444a28.png)

When they click the "Attempt Block Recovery" button, padding will be removed from the content, but everything else will work correctly.

The proposed code will now also check whether `spacingProps.style` exists.

---

By the way, I was doing a research on the state of `__experimentalGetSpacingClassesAndStyles()` before and could find the following thread: https://github.com/WordPress/gutenberg/issues/35920. There are no details shared unfortunately, but the function is being used with core Button block. Also, similar `getColorClassesAndStyles()` function is used with the Table and Pullquote blocks.